pkgs/sops-install-secrets/main.go

```diff
@@ -239,13 +240,25 @@ func recurseSecretKey(keys map[string]interface{}, wantedKey string) (string, er
 		if !ok {
 			return "", fmt.Errorf("The key '%s' cannot be found", keyUntilNow)
 		}
-		valWithWrongType, ok := val.(map[interface{}]interface{})
-		if !ok {
-			return "", fmt.Errorf("Key '%s' does not refer to a dictionary", keyUntilNow)
-		}
 		currentData = make(map[string]interface{})
-		for key, value := range valWithWrongType {
-			currentData[key.(string)] = value
+		// The 'if' here is to deal with key type discrepancy between YAML and
+		// JSON.  With YAML, it is 'interface {}'; with JSON, it is 'string'.
+		if format == Json {
```

Comment:
Looks good. Can you also extend one of our tests to have a nested key?

Reply:
@Mic92 What's the best way to do that? I don't really understand the test infrastructure in this repo. Especially what's going on with the Go tests. `nixos-tests.nix` seems more pratical; but how do you edit `secrets.json`? Running `nix run . pkgs/sops-install-secrets/test-assets/secrets.json` throws:

```
fingerprint: 26F82B82FDFFA024E08B9C8B67936C83AAC837D4
mv: cannot stat '/root/.gnupg': Permission denied
```